lib/collection/src/shards/local_shard/mod.rs

```diff
@@ -544,7 +544,7 @@ impl LocalShard {
         // (`SerdeWal::read_all` may even start reading WAL from some already truncated
         // index *occasionally*), but the storage can handle it.
 
-        for (op_num, update) in wal.read_all() {
+        for (op_num, update) in wal.read(wal.first_index()) {
```

Comment:
Why do we want to be more explicit here, isn't `read_all()` enough, fulfilling this purpose?

Reply:
Makes sense!

Yes, I'd vote for a `read_all(with_acknowledged)` which defines `first_index` or the other internally, returning back the doubly ended iterator.

Reply:
Tbh, I don't like these `with_acknowledged` methods. In my opinion, it makes *very* little difference with just doing `wal.read(wal.some_specific_index())`. Especially when we only ever use `read_all` *once* and always with `with_acknowledged = false`.

But I only *slightly* disagree with this, so I'm not too against `read_all(with_acknowledged)`.

Reply:
> Especially when we only ever use read_all once and always with with_acknowledged = false.

We don't.

When resolving the diff it would use `read_all(true)`.

Reply:
Yeah, I see now. Before we only used `read_all` in a single place. It makes sense now. 👍